File: Difficulty 3.3/Candy Division/CandyDivision.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;

public class CandyDivision{
    public static void main (String [] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long candies = Long.parseLong(br.readLine());
        long temp = candies -1;
        ArrayList<Long> list = new ArrayList<>();
        list.add((long) 0);
        list.add(candies - 1);
        for (int i = 2; i < temp; i++) {
            if (candies % i == 0) {
                long val = i;
                long val1 = candies / i;
                if (val != val1) {
                    list.add(val1 - 1);
                }
                list.add(val - 1);
            }
            temp = candies / i;
        }

        Collections.sort(list);
        for (int i = 0; i < list.size(); i++) {
            if (i != list.size() - 1) {
                System.out.print(list.get(i) + " ");
            } else {
                System.out.print(list.get(i));
            }
        }
        System.out.println();
    }
}

File: Difficulty 3.3/Candy Division/CandyDivision.py
n = int(input())
temp = n
factors = []
factors.append(0)
factors.append(n - 1)

i = 2
while (i < temp):
    if (n % i == 0):
        val = (i - 1)
        val1 = (n / i - 1)
        if (val != val1):
            factors.append(val1)
        factors.append(val)
        temp = (n / i)
    i += 1

factors.sort()

result = ""
for j in factors:
    result = result + str(j) + " "

print(result[0:len(result) - 1])
